tests/test_pre_commit.py

```diff
@@ -364,3 +372,183 @@ def test_wrap_markdown_cell(tmpdir):
     text = nb.cells[0].source
     assert len(text.splitlines()) >= 2
     assert text != long_text
+
+
+def test_is_untracked(tmpdir):
+    git = git_in_tmpdir(tmpdir)
+
+    # make a test file
+    file = tmpdir.join("test.txt")
+    file.write("test file\n")
+    file = str(file)
+
+    with tmpdir.as_cwd():
+        # untracked
+        assert is_untracked(file)
+
+        # added, not committed
+        git("add", file)
+        assert not is_untracked(file)
+
+        # committed
+        git("commit", "-m", "'test'")
+        assert not is_untracked(file)
+
+
+def test_ignore_unmatched_ignores(tmpdir):
+    # Unmatched file
+    file = tmpdir.join("test.txt")
+    file.write("Hello\n")
+
+    # Run jupytext
+    status = jupytext(
+        ["--from", "ipynb", "--to", "py:light", "--ignore-unmatched", str(file)]
+    )
+
+    assert status == 0
+    assert not tmpdir.join("test.py").exists()
+
+
+def test_alert_untracked_alerts(tmpdir):
+    git_in_tmpdir(tmpdir)
+
+    file = tmpdir.join("test.py")
+    file.write("print('hello')\n")
+
+    # Run jupytext
+    with tmpdir.as_cwd():
+        status = jupytext(
+            ["--from", ".py", "--to", "ipynb", "--alert-untracked", str(file)]
+        )
+
+    assert status != 0
+    assert tmpdir.join("test.ipynb").exists()
+
+
+def test_alert_untracked_not_alerts_for_tracked(tmpdir):
+    git = git_in_tmpdir(tmpdir)
+
+    # write test notebook
+    nb = new_notebook(cells=[new_markdown_cell("A short notebook")])
+    nb_file = str(tmpdir.join("test.ipynb"))
+    write(nb, nb_file)
+
+    # write existing output
+    tmpdir.join("test.py").write("# Hello")
+
+    # track output file
+    git("add", str("test.py"))
+
+    # Run jupytext
+    with tmpdir.as_cwd():
+        status = jupytext(
+            ["--from", "ipynb", "--to", "py:light", "--alert-untracked", str(nb_file)]
+        )
+
+    assert status == 0
+
+
+@requires_pre_commit
+def test_pre_commit_hook_for_new_file(tmpdir):
+    # get the path and revision of this repo, to use with pre-commit
+    repo_root = str(Path(__file__).parent.parent.resolve())
+    repo_rev = system("git", "rev-parse", "HEAD", cwd=repo_root).strip()
+
+    # set up the tmpdir repo with pre-commit
+    git = git_in_tmpdir(tmpdir)
+    sys = system_in_tmpdir(tmpdir)
+
+    pre_commit_config_yaml = dedent(
+        f"""
+        repos:
+        - repo: {repo_root}
+          rev: {repo_rev}
```

Comment:
Out of curiosity, is this a way to test the local version of the Jupytext hook? Well done!

Reply:
The local repo hook can be used for this: https://pre-commit.com/#advanced

Reply:
Yes it's to test the "current version", pre-commit will create a new env for it. So you know that what you are committing works